### PR TITLE
 ONS-UK-Economic-Accounts-balance-of-payments-current-account

### PR DESCRIPTION
**Pull request by osamede20**
Rework done on codes
Cubes class added

### CODE REVIEW

**Comment by osamede20** [comment]
I am not sure what is required of the main.py file. Running the main.py script would create the csv, metadata.trig and schema.json files of only the last script B7_B7A.py. This arises from the cubes class of the successive script overwriting the csv, trig and json files of the preceding script.

**Comment by osamede20** [comment]
The latter commit 550f657 is the valid one to be merged to Master

File: datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B2_B2A_B3_B3A.py [hunk 1]
```diff
@@ -117,59 +114,41 @@ def time2period(t):
...
 
-schema = CSVWMetadata('https://gss-cogs.github.io/family-trade/reference/')
-schema.create(destinationFolder / f'{OBS_ID}.csv', destinationFolder / f'{OBS_ID}.csv-schema.json')
+cubes.add_cube(scraper, tidy, title)
+cubes.output_all()
 
-tidy
+# + endofcell="--"
 
 # --
```
Comment (Shannon95) [inline datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B2_B2A_B3_B3A.py:154]: The same comments made on file B1.py apply here also.

File: datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B4_B4A_B4B.py [hunk 2]
```diff
@@ -97,68 +94,49 @@ def time2period(t):
...
+cubes.output_all()
 
-tidy
+# + endofcell="--"
 
 # --
-
-
-
-
```
Comment (Shannon95) [inline datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B4_B4A_B4B.py:164]: The same comments made on file B1.py apply here also.

File: datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B6B_B6B2_B6B3_B6C_B6C2_B6C3.py [hunk 3]
```diff
@@ -101,65 +98,44 @@ def time2period(t):
...
+cubes.output_all()
 
-tidy
+# + endofcell="--"
 
 # --
-
-
-
-
```
Comment (Shannon95) [inline datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B6B_B6B2_B6B3_B6C_B6C2_B6C3.py:165]: The same comments made on file B1.py apply here also.

File: datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B5_B5A.py [hunk 4]
```diff
@@ -90,70 +88,47 @@ def time2period(t):
...
-tidy
+# + endofcell="--"
 
 # --
-
-
-
-
-
-
```
Comment (Shannon95) [inline datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B5_B5A.py:159]: The same comments made on file B1.py apply here also.

File: datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B6_B6A.py [hunk 5]
```diff
@@ -92,67 +89,43 @@ def time2period(t):
...
-tidy
+# + endofcell="--"
 
 # --
-
-
-
-
-
-
```
Comment (Shannon95) [inline datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B6_B6A.py:158]: The same comments made on file B1.py apply here also.

File: datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B7_B7A.py [hunk 6]
```diff
@@ -88,64 +84,46 @@ def time2period(t):
...
-schema.create(destinationFolder / f'{OBS_ID}.csv', destinationFolder / f'{OBS_ID}.csv-schema.json')
+cubes.add_cube(scraper, tidy, title)
+cubes.output_all()
 
-tidy
+# + endofcell="--"
 
 # --
-
-
```
Comment (Shannon95) [inline datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/B7_B7A.py:151]: The same comments made on file B1.py apply here also.

File: datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/main.py [hunk 7]
```diff
@@ -1,36 +1,16 @@
...
-# %run "B5_B5A.py"
-
-# %run "B6_B6A.py"
-
-# %run "B6B_B6B2_B6B3_B6C_B6C2_B6C3.py"
-
-# %run "B7_B7A.py"
-
+for scripts = ['B1.py', 'B2_B2A_B3_B3A.py', 'B4_B4A_B4B.py', 'B5_B5A.py', 'B6_B6A.py', 'B6B_B6B2_B6B3_B6C_B6C2_B6C3.py', 'B7_B7A.py']: 
+    %run $script
```
Comment (Shannon95) [inline datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/main.py:16]: This doesn't actually run for me, I got the following error. 
```python
SyntaxError: invalid syntax
```

I used this instead and it worked for me: 
```python
import glob

py_files = [i for i in glob.glob('*.{}'.format('py'))]

for i in py_files:
    file = "'" + i + "'"
    if file.startswith("'main") == True:
        continue
    %run $file
```

**Review by Shannon95 — COMMENTED** [review]
Ideally, all code would be held within the main.py which makes this dataset a little harder to understand, I believe I originally wrote the code for this so I do apologies for over complicating it at the time. I pulled out each tab to transform separately in their own notebook to help my understanding at the start as there are 7 outputs. 

As we are updating to the new standards there are a few aspects to change/implement: 
- There are some changes needed around using the scraper, seed and selecting the latest distribution. 
- TransformTrace needs implementing across each notebook. 
- Previously each notebook had its own output as the dimensions were slightly different within each tab and could not easily be joined together into one dataframe. Therefore the title passed when adding the cube at the end of each needs to be different, the same titles as preciously used would be fine. You might need to have a convo with Mike about how this is correctly implemented using the Cubes() class.

File: datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/main.py (hunk 7 above)
Comment (osamede20) [reply datasets/ONS-UK-Economic-Accounts-balance-of-payments-current-account/main.py:16]: Thanks Shannon for reviewing the pull request. The feedback is noted and changes will be made shortly.

**Pull request merged**